src/staldates/joystick.py

```diff
@@ -188,14 +192,54 @@ def _update_camera(self):
             pass
 
 
-if __name__ == "__main__":
-    dev_str = "/dev/input/js1"
-    js = Joystick(dev_str)
-    cja = CameraJoystickAdapter(js)
+JOYSTICK_MAX = 32767
+JOYSTICK_HALF = JOYSTICK_MAX / 2
+
 
-    cam = VISCACamera("Camera 1", "/dev/ttyUSB0", 1)
-    cam.initialise()
+class SensitivityPrefsCameraJoystickAdapter(CameraJoystickAdapter):
+    def update_preferences(self):
+        CameraJoystickAdapter.update_preferences(self)
+        self._set_speed_params()
+
+    def set_camera(self, camera):
+        super(SensitivityPrefsCameraJoystickAdapter, self).set_camera(camera)
+        self._set_speed_params()
+
+    def _set_speed_params(self):
+        if self._camera:
+            self.max_pan = self.camera.maxPanSpeed
+            self.max_tilt = self.camera.maxTiltSpeed
+            self.min_zoom = self.camera.minZoomSpeed
+            self.max_zoom = self.camera.maxZoomSpeed
+        else:
+            self.max_pan = 0x18
+            self.max_tilt = 0x14
+            self.min_zoom = 2
+            self.max_zoom = 7
+
+        self.pan_sensitivity = Preferences.get('joystick.sensitivity.pan', 0.5)
+        self.tilt_sensitivity = Preferences.get('joystick.sensitivity.tilt', 0.5)
+        self.zoom_sensitivity = Preferences.get('joystick.sensitivity.zoom', 0.5)
+
+    def _interp(self, raw, max_value, sensitivity):
```

Comment:
I think this could be a static method as it does not touch `self` (it looks like you can add a @staticmethod decorator in python).

Being static with no side effects, it's very easily unit testable.

Also it would be good if it had a docstring.

Reply:
Rather than static, which is then hard to override in any future subclasses, I've pulled the method out on its own and passed it in as the default to the class constructor.